_includes/tutorials/multiple-event-types/kafka/markup/dev/make-multi-event-app.adoc

```diff
@@ -0,0 +1,147 @@
+The text assumes a method buildTopology exists and constructs the Kafka Streams application.  Feel free to modify the text below to suit your needs.
+////
+
+To complete this tutorial, you'll build an application that uses a `KafkaProducer` and  `KafkaConsumer` instance for producing both Avro and Protobuf.  The approach you'll take in this tutorial is not typicaly of applications you'll build in a production setting. But by using multiple clients you can compare how to handle multiple event types for each serialzer format.
+
+**To that end, the point of this sample application is this:** you want capture pageview and purchase events in the exact order that they occur and you feel the best option is to have these events produced to the same topic.  Since the customer id is the key, you are guaranteed to get these events in the order that they occur.
+
+
+
+Let's go over some of the key parts of the `KafkaMultiEventConsumerApplication` starting with the producer for the Protobuf events.
+
+NOTE: Since this an advanced topic, the tutorial doesn't go into the basics of using a `KafkaProducer`. For more details see the https://creating-first-apache-kafka-producer-application/confluent.html[KafkaProducer tutorial]
+
+[source, java]
+.KafkaProducer for Protobuf events
+----
+public void produceProtobufEvents(final Supplier<Producer<String, CustomerEventProto.CustomerEvent>> producerSupplier,
+                                  final String topic,
+                                  final List<CustomerEventProto.CustomerEvent> protoCustomerEvents) {
+
+        try (Producer<String, CustomerEventProto.CustomerEvent> producer = producerSupplier.get()) { <1>
+            protoCustomerEvents.stream()    <2>
+                    .map((event -> new ProducerRecord<>(topic, event.getId(), event))) <3>
+                    .forEach(producerRecord -> producer.send(producerRecord, ((metadata, exception)
+
+      //Details left out for clarity
+
+   // Relevant configurations
+
+ protoProduceConfigs.put(ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG, KafkaProtobufSerializer.class);
+}
+----
+
+<1> Retrieving the producer instance from the `Supplier`
+<2> Using a `java.util.stream` to map each event into  a `ProducerRecord` then send them to the broker
+<3> Creating the `ProducerRecord` instance.
+
+There are two points to emphasize here.  The first is the type of the producer, its using `CustomerEventProto.CustomerEvent` since you must use an outer class with Protobuf, the generics on the producer are a concrete type.  As a result, to set the key to be the customer id you can call the `CustomerEvent#getId` method directly.  Note the use of the `Supplier` to provide the producer this is done to delay the instantiation until the `Supplier.get()` method is executed.  Using a supplier also makes testing easier by simplifying the process of providing a different implementation.
```

Comment:
```suggestion
There are two points to emphasize here.  The first is the type of the producer, it's using `CustomerEventProto.CustomerEvent` since you must use an outer class with Protobuf, the generics on the producer are a concrete type.  As a result, to set the key to be the customer id you can call the `CustomerEvent#getId` method directly.  Note the use of the `Supplier` to provide the producer this is done to delay the instantiation until the `Supplier.get()` method is executed.  Using a supplier also makes testing easier by simplifying the process of providing a different implementation.
```

Reply:
Is this a run-on sentence?